### DISCUSSION TITLE
MQTT?

### DISCUSSION THREAD

**Discussion by nborson** [rasclatt-dot-com/ESP32-Plant-Waterer-for-ESPHome#4] [Q&A]
Thanks for this great introduction; I have my first waterer up and running, I think, by (mostly) following this example.
The hardware suggestion is excellent, inexpensive and so easy to install!
I don't understand why you use both [MQTT](https://esphome.io/components/mqtt.html) and the ESPHome [native API](https://esphome.io/components/api.html). Perhaps you use MQTT to communicate with other parts of a home automation system (besides Home Assistant)? Or that was the way to do it two years ago? I don't have some of the failsafe and dashboard parts set up yet, but the basic automation seems to work without MQTT, which I don't use otherwise and seems needlessly complex.
Again, thanks! 👍🙏😁

**Comment by SR-G** [comment]
Hey.
I wondered the same.
I'm pretty sure the MQTT metrics are only used for the "number of waterings over the last 4H" counter (per the HA configuration.yaml config to be added, this metric is computed from MQTT).
It's probably possible to compute this outside MQTT (but i'm not at all a HA expert, so i'm not able to see how).
And i'm pretty sure that if that counter was computed without MQTT, then the MQTT push could be removed (which would be probably better, as it would make the whole setup simpler and easier to understand / update).
